### PR TITLE
[TypeSpecGenerate.ps1] Run TypeSpec using "node entrypoints/cli.js"

### PR DESCRIPTION
**Pull request by mikeharder**
- @typespec/compiler@0.46 changes the file layout inside the package, so "dist/core/cli/cli.js" is no longer valid
- Path "entrypoint/tsp.js" is much less likely to change in future releases

### CODE REVIEW

File: eng/TypeSpecGenerate.ps1 [hunk 1]
```diff
@@ -23,7 +23,7 @@ foreach ($directory in Get-ChildItem $testEmitterPath -Directory)
     {
         Remove-Item $projectPath/Generated -Force -Recurse
     }
-    node node_modules/@typespec/compiler/dist/core/cli/cli.js compile $projectPath/$testName.tsp --emit @azure-tools/typespec-csharp --option @azure-tools/typespec-csharp.emitter-output-dir=$projectPath --option @azure-tools/typespec-csharp.skipSDKGeneration=true --option @azure-tools/typespec-csharp.save-inputs=true
+    npx --no tsp compile $projectPath/$testName.tsp --emit @azure-tools/typespec-csharp --option @azure-tools/typespec-csharp.emitter-output-dir=$projectPath --option @azure-tools/typespec-csharp.skipSDKGeneration=true --option @azure-tools/typespec-csharp.save-inputs=true
```
Comment (mikeharder) [inline eng/TypeSpecGenerate.ps1:26]: If this doesn't work, we should be able to use this instead:

```
node node_modules/@typespec/compiler/entrypoints/cli.js
```

Comment (mikeharder) [inline eng/TypeSpecGenerate.ps1:26]: If this doesn't work, we should be able to use this instead:


```suggestion
    node node_modules/@typespec/compiler/entrypoints/cli.js compile $projectPath/$testName.tsp --emit @azure-tools/typespec-csharp --option @azure-tools/typespec-csharp.emitter-output-dir=$projectPath --option @azure-tools/typespec-csharp.skipSDKGeneration=true --option @azure-tools/typespec-csharp.save-inputs=true
```

**Review by m-nash — CHANGES_REQUESTED** [review]
we need to validate how this plays with other versions installed on the same machine

**Comment by mikeharder** [comment]
> we need to validate how this plays with other versions installed on the same machine

Seems to work correctly on my dev machine:

```
$ npm ls -g
/home/mharder/.nvm/versions/node/v16.20.1/lib
├── @typespec/compiler@0.45.2
├── corepack@0.17.0
└── npm@8.19.4

$ npm ls
tsp-local-global@ /home/mharder/tmp/tsp-local-global
└── @typespec/compiler@0.46.0-dev.30

$ tsp --version
TypeSpec compiler v0.46.0-dev.30
0.46.0-dev.30

$ npx --no tsp -- --version
TypeSpec compiler v0.46.0-dev.30
0.46.0-dev.30
```

File: eng/TypeSpecGenerate.ps1 [hunk 2]
```diff
@@ -47,7 +47,7 @@ foreach ($directory in Get-ChildItem $samplePath -Directory)
      {
          Remove-Item $projectPath/Generated -Force -Recurse
      }
-    node node_modules/@typespec/compiler/dist/core/cli/cli.js compile $projectPath/$testName.tsp --emit @azure-tools/typespec-csharp --option @azure-tools/typespec-csharp.emitter-output-dir=$projectPath --option @azure-tools/typespec-csharp.skipSDKGeneration=true --option @azure-tools/typespec-csharp.save-inputs=true
+    npx --no tsp compile $projectPath/$testName.tsp --emit @azure-tools/typespec-csharp --option @azure-tools/typespec-csharp.emitter-output-dir=$projectPath --option @azure-tools/typespec-csharp.skipSDKGeneration=true --option @azure-tools/typespec-csharp.save-inputs=true
```
Comment (mikeharder) [inline eng/TypeSpecGenerate.ps1:50]:
```suggestion
    node node_modules/@typespec/compiler/cmd/tsp.js compile $projectPath/$testName.tsp --emit @azure-tools/typespec-csharp --option @azure-tools/typespec-csharp.emitter-output-dir=$projectPath --option @azure-tools/typespec-csharp.skipSDKGeneration=true --option @azure-tools/typespec-csharp.save-inputs=true
```

**Review by m-nash — APPROVED** [review]

**Pull request merged**